pkg/kmm/kernelmapping.go

```diff
@@ -73,6 +57,38 @@ func (builder *KernelMappingBuilder) BuildKernelMappingConfig() (*moduleV1Beta1.
 	return builder.definition, nil
 }
 
+func (builder *KernelMappingBuilder) WithRegexp(regexp string) *KernelMappingBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Adding regexp %s to kernel mapping", regexp)
+
+	builder.definition.Regexp = regexp
+
+	if regexp == "" {
```

Comment:
Please move this validation before: 
builder.definition.Regexp = regexp
Update Literal only if builder.errorMsg == ""

Reply:
The only reason I didn't do it to begin with is because this is how the [original-function](https://github.com/openshift-kni/eco-goinfra/blob/3c31a5ef4dea4b9dd46d392f9a15dff39672e058/pkg/kmm/kernelmapping.go#L29-L39) was written and I didn't want to change the logic.

Should I still modify it? I don't see any CI in the repo and I am not sure who else uses those functions.